scripts/gen_idt.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import sys
+import struct
+import os
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import SymbolTableSection
+
+# This will never change, first selector in the GDT after the null selector
+KERNEL_CODE_SEG = 0x08
+
+# These exception vectors push an error code onto the stack.
+ERR_CODE_VECTORS = [8, 10, 11, 12, 13, 14, 17]
+
+def debug(text):
+    if not args.verbose:
+        return
+    sys.stdout.write(os.path.basename(sys.argv[0]) + ": " + text + "\n")
+
+def error(text):
+    sys.stderr.write(os.path.basename(sys.argv[0]) + ": " + text + "\n")
+    sys.exit(1)
+
+irq_gate_desc_format = "<HHBBH"
```

Comment:
Style: Similar comment, and a repeat of earlier one: this string isn't a "constant", it's code to be interpreted at runtime.  It should go where it's used as a literal, not here.